File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.js
function solution(num_list) {
    return num_list.reduce((acc, cur) => {
        let count = 0;
        
        while (cur !== 1) {
            if (cur % 2 === 1) {
                cur = (cur - 1) / 2;
            } else {
                cur /= 2;
            }
            count++;
        }
        return acc + count;
    }, 0);
}

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int result = 0;
        
        for (int n : num_list) {
            result += func(n);
        }
        
        return result;
    }
    
    private int func(int num) {
        int cnt = 0;
        
        while (num != 1) {
            if (num % 2 == 0) {
                num /= 2;
            } else {
                num = (num - 1) / 2;
            }
            cnt++;
        }
        
        return cnt;
    }
}